src/main/groovy/com/athaydes/spockframework/report/template/TemplateReportCreator.groovy

```diff
@@ -79,7 +79,7 @@ class TemplateReportCreator implements IReportCreator {
 
     def createFeaturesCallback( SpecData data ) {
         return [ eachFeature: { Closure callback ->
-            for ( feature in data.info.allFeatures ) {
+            for ( feature in data.info.allFeaturesInExecutionOrder ) {
```

Comment:
Done. Btw, I Just noticed that your `TemplateReportCreatorSpec` is currently failing, independent of this change.